src/reynir_correct/errtokenizer.py

```diff
@@ -1148,7 +1157,7 @@ def get() -> CorrectToken:
                         )
                     )
                     yield token
-                else:
+                elif "C001" not in ignore_rules:
```

Comment:
Changed the if flow, so that ignore_rules checks are a specific if statement where possible. This should ensure we catch the error token in appropriate places and then decide whether we want to annotate it, instead of the error token skipping that if statement since it doesn't fulfill all conditions and being handled in other if statements, which can lead to unforeseen consequences.